File: sdk/r/R/ai_factory.R
#' HiTechCloud AI Factory: templates, instances, SSH keys, volumes, clusters
#' @param http HttpClient object
#' @return Resource object
#' @export
AiFactoryResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "AiFactoryResource"
  env
}

#' GET /api/service/{id}/templates - List templates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/templates", list(...))
}

#' POST /api/service/{id}/templates - Save a template
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
saveTemplate <- function(self, ...) {
  self$http$post("/api/service/{id}/templates", list(...))
}

#' GET /api/service/{id}/hitechcloud/mode - Get provisioning mode
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getProvisioningMode <- function(self, ...) {
  self$http$get("/api/service/{id}/hitechcloud/mode", list(...))
}

#' GET /api/service/{id}/hitechcloud/examples - Get JSON examples
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getJsonExamples <- function(self, ...) {
  self$http$get("/api/service/{id}/hitechcloud/examples", list(...))
}

#' GET /api/service/{id}/hitechcloud/schema - Get API schema
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getApiSchema <- function(self, ...) {
  self$http$get("/api/service/{id}/hitechcloud/schema", list(...))
}

#' GET /api/service/{id}/instances - List all GPU instances
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listInstances <- function(self, ...) {
  self$http$get("/api/service/{id}/instances", list(...))
}

#' POST /api/service/{id}/instances - Create a GPU instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instances", list(...))
}

#' GET /api/service/{id}/instances/types - List available GPU instance types
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listInstanceTypes <- function(self, ...) {
  self$http$get("/api/service/{id}/instances/types", list(...))
}

#' GET /api/service/{id}/instances/{instance_id} - Get instance details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getInstance <- function(self, ...) {
  self$http$get("/api/service/{id}/instances/{instance_id}", list(...))
}

#' POST /api/service/{id}/instances/{instance_id}/update - Update instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
updateInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instances/{instance_id}/update", list(...))
}

#' POST /api/service/{id}/instances/{instance_id}/restart - Restart instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
restartInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instances/{instance_id}/restart", list(...))
}

#' POST /api/service/{id}/instances/{instance_id}/delete - Delete instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instances/{instance_id}/delete", list(...))
}

#' GET /api/service/{id}/instance - Get service instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getServiceInstance <- function(self, ...) {
  self$http$get("/api/service/{id}/instance", list(...))
}

#' POST /api/service/{id}/instance/sync - Synchronize service resource
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
syncServiceInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instance/sync", list(...))
}

#' POST /api/service/{id}/instance/restart - Restart service instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
restartServiceInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instance/restart", list(...))
}

#' POST /api/service/{id}/instance/update - Update service instance
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
updateServiceInstance <- function(self, ...) {
  self$http$post("/api/service/{id}/instance/update", list(...))
}

#' GET /api/service/{id}/sshkeys - List SSH keys
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listSshKeys <- function(self, ...) {
  self$http$get("/api/service/{id}/sshkeys", list(...))
}

#' POST /api/service/{id}/sshkeys - Add SSH key
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
addSshKey <- function(self, ...) {
  self$http$post("/api/service/{id}/sshkeys", list(...))
}

#' GET /api/service/{id}/sshkeys/{key_id} - Get SSH key details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getSshKey <- function(self, ...) {
  self$http$get("/api/service/{id}/sshkeys/{key_id}", list(...))
}

#' POST /api/service/{id}/sshkeys/{key_id}/delete - Delete SSH key
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteSshKey <- function(self, ...) {
  self$http$post("/api/service/{id}/sshkeys/{key_id}/delete", list(...))
}

#' POST /api/service/{id}/sshkeys/{key_id}/setdefault - Set default SSH key
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
setDefaultSshKey <- function(self, ...) {
  self$http$post("/api/service/{id}/sshkeys/{key_id}/setdefault", list(...))
}

#' GET /api/service/{id}/volumes - List volumes
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listVolumes <- function(self, ...) {
  self$http$get("/api/service/{id}/volumes", list(...))
}

#' POST /api/service/{id}/volumes - Create a storage volume
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createVolume <- function(self, ...) {
  self$http$post("/api/service/{id}/volumes", list(...))
}

#' GET /api/service/{id}/volumes/types - List volume types
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listVolumeTypes <- function(self, ...) {
  self$http$get("/api/service/{id}/volumes/types", list(...))
}

#' GET /api/service/{id}/volumes/{volume_id} - Get volume details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getVolume <- function(self, ...) {
  self$http$get("/api/service/{id}/volumes/{volume_id}", list(...))
}

#' POST /api/service/{id}/volumes/{volume_id}/delete - Delete volume
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteVolume <- function(self, ...) {
  self$http$post("/api/service/{id}/volumes/{volume_id}/delete", list(...))
}

#' GET /api/service/{id}/templates/featured - List featured templates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listFeaturedTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/templates/featured", list(...))
}

#' GET /api/service/{id}/templates/{template_id} - Get template details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getTemplate <- function(self, ...) {
  self$http$get("/api/service/{id}/templates/{template_id}", list(...))
}

#' POST /api/service/{id}/templates/{template_id}/update - Update template
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
updateTemplate <- function(self, ...) {
  self$http$post("/api/service/{id}/templates/{template_id}/update", list(...))
}

#' POST /api/service/{id}/templates/{template_id}/delete - Delete template
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteTemplate <- function(self, ...) {
  self$http$post("/api/service/{id}/templates/{template_id}/delete", list(...))
}

#' GET /api/service/{id}/clusters - List GPU clusters
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listClusters <- function(self, ...) {
  self$http$get("/api/service/{id}/clusters", list(...))
}

#' POST /api/service/{id}/clusters - Create a GPU cluster
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
createCluster <- function(self, ...) {
  self$http$post("/api/service/{id}/clusters", list(...))
}

#' GET /api/service/{id}/clusters/types - List available GPU cluster types
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listClusterTypes <- function(self, ...) {
  self$http$get("/api/service/{id}/clusters/types", list(...))
}

#' GET /api/service/{id}/cluster - Get service cluster
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getServiceCluster <- function(self, ...) {
  self$http$get("/api/service/{id}/cluster", list(...))
}

#' GET /api/service/{id}/clusters/{cluster_id} - Get cluster details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getCluster <- function(self, ...) {
  self$http$get("/api/service/{id}/clusters/{cluster_id}", list(...))
}

#' POST /api/service/{id}/clusters/{cluster_id}/delete - Delete cluster
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteCluster <- function(self, ...) {
  self$http$post("/api/service/{id}/clusters/{cluster_id}/delete", list(...))
}

File: sdk/c/src/ai_factory.c
#include "hitechcloud/ai_factory.h"
#include <stdlib.h>


/* GET /api/service/{id}/templates - List templates */
char *hc_ai_factory_list_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/templates", NULL);
}

/* POST /api/service/{id}/templates - Save a template */
char *hc_ai_factory_save_template(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/templates", NULL);
}

/* GET /api/service/{id}/hitechcloud/mode - Get provisioning mode */
char *hc_ai_factory_get_provisioning_mode(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/hitechcloud/mode", NULL);
}

/* GET /api/service/{id}/hitechcloud/examples - Get JSON examples */
char *hc_ai_factory_get_json_examples(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/hitechcloud/examples", NULL);
}

/* GET /api/service/{id}/hitechcloud/schema - Get API schema */
char *hc_ai_factory_get_api_schema(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/hitechcloud/schema", NULL);
}

/* GET /api/service/{id}/instances - List all GPU instances */
char *hc_ai_factory_list_instances(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/instances", NULL);
}

/* POST /api/service/{id}/instances - Create a GPU instance */
char *hc_ai_factory_create_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instances", NULL);
}

/* GET /api/service/{id}/instances/types - List available GPU instance types */
char *hc_ai_factory_list_instance_types(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/instances/types", NULL);
}

/* GET /api/service/{id}/instances/{instance_id} - Get instance details */
char *hc_ai_factory_get_instance(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/instances/{instance_id}", NULL);
}

/* POST /api/service/{id}/instances/{instance_id}/update - Update instance */
char *hc_ai_factory_update_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instances/{instance_id}/update", NULL);
}

/* POST /api/service/{id}/instances/{instance_id}/restart - Restart instance */
char *hc_ai_factory_restart_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instances/{instance_id}/restart", NULL);
}

/* POST /api/service/{id}/instances/{instance_id}/delete - Delete instance */
char *hc_ai_factory_delete_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instances/{instance_id}/delete", NULL);
}

/* GET /api/service/{id}/instance - Get service instance */
char *hc_ai_factory_get_service_instance(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/instance", NULL);
}

/* POST /api/service/{id}/instance/sync - Synchronize service resource */
char *hc_ai_factory_sync_service_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instance/sync", NULL);
}

/* POST /api/service/{id}/instance/restart - Restart service instance */
char *hc_ai_factory_restart_service_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instance/restart", NULL);
}

/* POST /api/service/{id}/instance/update - Update service instance */
char *hc_ai_factory_update_service_instance(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/instance/update", NULL);
}

/* GET /api/service/{id}/sshkeys - List SSH keys */
char *hc_ai_factory_list_ssh_keys(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/sshkeys", NULL);
}

/* POST /api/service/{id}/sshkeys - Add SSH key */
char *hc_ai_factory_add_ssh_key(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/sshkeys", NULL);
}

/* GET /api/service/{id}/sshkeys/{key_id} - Get SSH key details */
char *hc_ai_factory_get_ssh_key(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/sshkeys/{key_id}", NULL);
}

/* POST /api/service/{id}/sshkeys/{key_id}/delete - Delete SSH key */
char *hc_ai_factory_delete_ssh_key(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/sshkeys/{key_id}/delete", NULL);
}

/* POST /api/service/{id}/sshkeys/{key_id}/setdefault - Set default SSH key */
char *hc_ai_factory_set_default_ssh_key(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/sshkeys/{key_id}/setdefault", NULL);
}

/* GET /api/service/{id}/volumes - List volumes */
char *hc_ai_factory_list_volumes(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/volumes", NULL);
}

/* POST /api/service/{id}/volumes - Create a storage volume */
char *hc_ai_factory_create_volume(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/volumes", NULL);
}

/* GET /api/service/{id}/volumes/types - List volume types */
char *hc_ai_factory_list_volume_types(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/volumes/types", NULL);
}

/* GET /api/service/{id}/volumes/{volume_id} - Get volume details */
char *hc_ai_factory_get_volume(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/volumes/{volume_id}", NULL);
}

/* POST /api/service/{id}/volumes/{volume_id}/delete - Delete volume */
char *hc_ai_factory_delete_volume(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/volumes/{volume_id}/delete", NULL);
}

/* GET /api/service/{id}/templates/featured - List featured templates */
char *hc_ai_factory_list_featured_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/templates/featured", NULL);
}

/* GET /api/service/{id}/templates/{template_id} - Get template details */
char *hc_ai_factory_get_template(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/templates/{template_id}", NULL);
}

/* POST /api/service/{id}/templates/{template_id}/update - Update template */
char *hc_ai_factory_update_template(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/templates/{template_id}/update", NULL);
}

/* POST /api/service/{id}/templates/{template_id}/delete - Delete template */
char *hc_ai_factory_delete_template(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/templates/{template_id}/delete", NULL);
}

/* GET /api/service/{id}/clusters - List GPU clusters */
char *hc_ai_factory_list_clusters(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/clusters", NULL);
}

/* POST /api/service/{id}/clusters - Create a GPU cluster */
char *hc_ai_factory_create_cluster(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/clusters", NULL);
}

/* GET /api/service/{id}/clusters/types - List available GPU cluster types */
char *hc_ai_factory_list_cluster_types(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/clusters/types", NULL);
}

/* GET /api/service/{id}/cluster - Get service cluster */
char *hc_ai_factory_get_service_cluster(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/cluster", NULL);
}

/* GET /api/service/{id}/clusters/{cluster_id} - Get cluster details */
char *hc_ai_factory_get_cluster(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/clusters/{cluster_id}", NULL);
}

/* POST /api/service/{id}/clusters/{cluster_id}/delete - Delete cluster */
char *hc_ai_factory_delete_cluster(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/clusters/{cluster_id}/delete", NULL);
}

